crates/bevy_ecs/src/system/commands/command_queue.rs

```diff
@@ -157,7 +157,7 @@ impl CommandQueue {
             // SAFETY: The data underneath the cursor must correspond to the type erased in metadata,
             // since they were stored next to each other by `.push()`.
             // For ZSTs, the type doesn't matter as long as the pointer is non-null.
-            let size = unsafe { (meta.consume_command_and_get_size)(cmd, &mut world) };
+            let size = unsafe { (meta.consume_command_and_get_size)(cmd, world.as_deref_mut()) };
```

Comment:
Great catch on this, though I wonder if the compiler will lift the branch out of the loop.

Reply:
This seems to remove two `mov` instructions from outside of the hot loop: https://github.com/james7132/bevy_asm_tests/compare/main...remove-double-deref#diff-8e36b64d5c51682bb463a51ec6e670b2ec99545945bc4753f4e77ef274c5faa0L27. This likely has a stronger impact inside of the commands themselves.

Reply:
I don't think this introduces a branch -- `as_deref_mut()` should be a no-op here

Reply:
I bet the moves getting removed are due to the fact that the compiler no longer has to deal with the possibility of a command swapping the world from `None` to `Some` or vice versa.